zcash.go

```diff
@@ -0,0 +1,42 @@
+package coincodec
+
+import (
+	"errors"
+	"github.com/wealdtech/go-slip44"
+)
+
+const (
+	ZCashAddressLength = 22
+	ZCashStaticPrefix = 0x1c
+)
+
+func init() {
+	toBytesMap[slip44.ZCASH] = ZCashDecodeToBytes
+	toStringMap[slip44.ZCASH] = ZCashEncodeToString
+}
+
+// ZCashDecodeToBytes converts the input string to a byte array
+func ZCashDecodeToBytes(input string) ([]byte, error) {
+	decoded, err := Base58ChecksumDecode(input, Base58DefaultAlphabet)
+	if err != nil {
+		return nil, err
+	}
+	if len(decoded) != ZCashAddressLength {
+		return nil, errors.New("Invalid length")
+	}
+	// prefix check
+	staticPrefix := decoded[0]
```

Comment:
Zcash has two bytes prefix, encoded as t1 or t3

Reply:
I based on walletcore implementation, and there is no prefix check there at all.  It accepts this as valid: "t488N47kTBknqPow5zKQbZjKTGKRpJkh4ab".  Should I check here to accept t1 and t3 only?  We don't need to be very strict here, important is that ENS produces the same input.

Reply:
hmm, then looks like it's a wallet core bug